code/__HELPERS/unsorted.dm

```diff
@@ -683,23 +683,23 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	var/holding = user.get_active_hand()
 	var/timefraction = round(time/numticks)
 	var/image/progbar
+	var/continue_looping = 1
 	for(var/i = 1 to numticks)
 		if(user.client && progress)
 			progbar = make_progress_bar(i, numticks, target)
-			user.client.images |= progbar
+			assign_progress_bar(user, progbar)
 		sleep(timefraction)
 		if(!user || !target)
-			if(user && user.client)
-				user.client.images -= progbar
-			return 0
-		if (!uninterruptible && (user.loc != user_loc || target.loc != target_loc || user.get_active_hand() != holding || user.incapacitated() || user.lying ))
-			if(user && user.client)
-				user.client.images -= progbar
-			return 0
-		if(user && user.client)
-			user.client.images -= progbar
-	if(user && user.client)
-		user.client.images -= progbar
+			continue_looping = 0
+			
+		if (continue_looping && !uninterruptible && (user.loc != user_loc || target.loc != target_loc || user.get_active_hand() != holding || user.incapacitated() || user.lying ))
+			continue_looping = 0
+		
+		cancel_progress_bar(user, progbar)//Clear the way for the next progbar image
+		if(!continue_looping)
+			return continue_looping

```

Comment:
fair enough. I need to squash the commits and tidy this up, so I'll include it as part of that.